roles/katello_client/tasks/main.yml

```diff
@@ -1,18 +1,42 @@
 ---
+- name: 'Cleanup previous registrations'
+  include_tasks: cleanup.yml
+  when: katello_client_cleanup
+
 - name: 'Install bootstrap rpm'
   yum:
     name: "http://{{ katello_client_server }}/pub/katello-ca-consumer-latest.noarch.rpm"
     state: "present"
 
-- name: 'Register client with subscription-manager'
+- name: 'Register client with subscription-manager as user {{ katello_client_username }}'
   redhat_subscription:
     state: "present"
     org_id: "{{ katello_client_organization }}"
     environment: "{{ katello_client_environment }}"
     username: "{{ katello_client_username }}"
     password: "{{ katello_client_password }}"
+  when:
+    - katello_client_activationkey == ""
+
+- name: 'Register client with subscription-manager using AK {{ katello_client_activationkey }}'
+  redhat_subscription:
+    state: "present"
+    org_id: "{{ katello_client_organization }}"
+    activationkey: "{{ katello_client_activationkey }}"
+  when:
+    - katello_client_activationkey != ""
+
+- name: 'Enable extra repositories'
+  command: subscription-manager repos --enable {{ item }}
+  with_items: "{{ katello_client_extra_repos }}"
+
+- name: 'Install katello-host-tools-fact-plugin'
+  yum:
+    name: "katello-host-tools-fact-plugin"
```

Comment:
I am curious the opt here for katello-host-tools-fact-plugin instead of the base package of katello-host-tools (which requires the fact plugin) ?

Do you think adding an additional option for katello-host-tools-tracer would be useful?

Reply:
that's just me being silly, k-h-t is of course the right package here.

will add a tracer option too